common/SmokeTests/SmokeTest/Program.cs

```diff
@@ -1,37 +1,278 @@
+    LogInformation($"Processing assembly: [{ assembly.FullName }]...");
+
+    var typeCount = 0;
+
+    try
+    {
+        // Process each type in the assembly.  The call to "ProcessType" is safe and
+        // will not throw.
+
+        foreach (var type in assembly.GetTypes())
+        {
+            ++typeCount;
+            ProcessType(type);
+        }
+    }
+    catch (ReflectionTypeLoadException) when (assembly.FullName.StartsWith("System."))
     {
-        static async Task<int> Main(string[] args)
+        // Not expected, but not impactful.  System assemblies aren't indicative of
+        // transitive dependency conflicts.
+    }
+    catch (ReflectionTypeLoadException ex)
+    {
+        // Not expected; this typically indicates an issue with loading an assembly
+        // and may indicate a version conflict with transitive dependencies.
+
+        var builder = new StringBuilder($"{Environment.NewLine }[{ assembly.FullName }] could not load types for inspection.  Error: { ex }{ Environment.NewLine } \tLoad Error Details:{ Environment.NewLine }");
+
+        foreach (var loadEx in ex.LoaderExceptions)
         {
-            Console.WriteLine("SMOKE TEST FOR TRACK 2 LIBRARIES");
-
-            await KeyVaultTest.RunTests();
-            await BlobStorageTest.RunTests();
-            await EventHubsTest.RunTests();
-            // Temporarily disable Cosmos DB smoke tests
-            // Issue to re-enable:
-            // https://github.com/Azure/azure-sdk-for-net/issues/11297
-            // await CosmosDBTest.RunTests();
-
-            Console.WriteLine();
-            Console.WriteLine("LIBRARY TESTS COMPLETE");
-            Console.WriteLine();
-            Console.WriteLine("SMOKE TEST FOR TRACK 2 SAMPLES");
-
-            await IotHubConnectionTests.RunTests();
-
-            Console.WriteLine("SAMPLES TESTS COMPLETE");
-            Console.WriteLine();
-            return 0;
+            builder.Append("\t\t - ");
+            builder.AppendLine(loadEx.Message);
         }
+
+        builder.Append(Environment.NewLine);
+
+        ReportError();
+        LogError(builder.ToString());
+    }
+    catch (Exception ex)
+    {
+        ReportError();
+        LogError($"{Environment.NewLine }[{ assembly.FullName }] could not be fully processed.  Error: { ex }{ Environment.NewLine }");
+    }
+
+    totalTypeCount += typeCount;
+    LogInformation($"Completed assembly: [{ assembly.FullName }].  {typeCount:n0} types inspected.{ Environment.NewLine }");
+}
+
+// End the test run and report on the state.
+
+LogInformation($"{ Environment.NewLine }Smoke Test run complete.  Total types inspected: {totalTypeCount:n0}");
+
+if (returnCode == 0)
+{
+    LogInformation("Passed.", ConsoleColor.Green);
+}
+else
+{
+    LogInformation("Failed.", ConsoleColor.Red);
+}
+
+return returnCode;
+
+// Local function definitions
+
+void ReportError() => returnCode = -1;
+
+void LogError(string message, ConsoleColor? color = default)
```

Comment:
We should add an option that will write this error in the magic DevOps way so they will be prompted to the error list in the pipeline to allow for easier finding. See https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/logging.ps1#L19 where we do this in PS. All we need to do is check for the SYSTEM_TEAMPROJECTID env variable and if it is set format the output with the magic header.

Reply:
I'll work this in.